src/Sparrow/AsyncManualResetEvent.cs

```diff
@@ -67,19 +62,6 @@ public Task<bool> WaitAsync()
             [Pure]
             public async Task<bool> WaitAsync(TimeSpan timeout)
             {
-                Debug.Assert(timeout != TimeSpan.MaxValue);
```

Comment:
It was added since passing such value to `Task.Delay` resulted in throwing an exception